cpp/src/Ice/TcpEndpointI.cpp

```diff
@@ -122,6 +122,26 @@ IceInternal::TcpEndpointI::datagram() const
     return false;
 }
 
+shared_ptr<EndpointI>
+IceInternal::TcpEndpointI::toPublishedEndpoint(string publishedHost) const
+{
+    if (publishedHost.empty() && !isAddressValid(_sourceAddr) && _connectionId.empty())
```

Comment:
I read this as:

If the published host is empty and there is no local options. The published endpoint is the actual endpoint. Otherwise replace publishedHost and remove local options.

Maybe add a comment to clarify.

Reply:
See latest version.

An empty publishedHost means "keep existing host as is".